### PR TITLE
KOGITO-2246: Add keycloak integration at quarkus management console App

### PR DESCRIPTION
**Pull request by nmirasch**
https://issues.redhat.com/browse/KOGITO-2246

Adds Management Console quarkus application properties to enable and configure Keycloak authentication
Added a KeycloakClient  to encapsulate all the interaction with Keycloak

### CODE REVIEW

**Comment by nmirasch** [comment]
To try this just have to create the keycloak docker image and include the kogito-realm executing:

docker run -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin  -e KEYCLOAK_IMPORT=/tmp/kogito-realm.json -v {absolute_path}/kogito-apps/config/kogito-realm.json:/tmp/kogito-realm.json -p 8280:8080  jboss/keycloak

and start the management console at port 8380, (the keycloak client 'kogito-console' is configured to use port) and enabling auth
mvn clean compile quarkus:dev -Dquarkus.http.port=8380 -Dkogito.auth.enabled=true

To change any of this client configuration access to http://localhost:8280/auth/admin/master/console/#/realms/kogito with admin/admin

I hope this helps!

**Review by pefernan — APPROVED** [review]
@nmirasch yea works perfect! Thank you for the instructions, should he have them on the README?

File: management-console/src/main/java/org/kie/kogito/mgmt/VertxRouter.java [hunk 1]
```diff
@@ -57,5 +77,8 @@ public void handle(RoutingContext context) {
         }
     }
 
+    protected String getResource(){
```
Comment (cristianonicolai) [inline management-console/src/main/java/org/kie/kogito/mgmt/VertxRouter.java]: unsued?

File: management-console/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+kogito.auth.enabled=false
```
Comment (cristianonicolai) [inline management-console/src/main/resources/application.properties]: @nmirasch do we need to redefine these properties here, but rely on default values instead?

**Review by cristianonicolai — COMMENTED** [review]
+1 for adding instructions to README

**Review by cristianonicolai — APPROVED** [review]

File: config/kogito-realm.json [hunk 3]
```diff
@@ -800,6 +800,63 @@
         "microprofile-jwt"
       ]
     },
+    {
+      "clientId": "kogito-quarkus-console",
```
Comment (Sgitario) [inline config/kogito-realm.json]: why "kogito-quarkus-console"? It sounds weird to me adding "quarkus", maybe just "kogito-console" or "kogito-management-console"?

File: ui-packages/packages/task-console/src/index.tsx [hunk 4]
```diff
@@ -3,17 +3,16 @@ import ReactDOM from 'react-dom';
...
 import { ApolloProvider } from 'react-apollo';
-import Keycloak from 'keycloak-js';
-import axios from 'axios';
 import '@patternfly/patternfly/patternfly.css';
 import { Nav, NavList, NavItem } from '@patternfly/react-core';
 import { HttpLink } from 'apollo-link-http';
 import { onError } from 'apollo-link-error';
 import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
-import { ServerUnavailable } from '@kogito-apps/common';
+import {appRenderWithAxiosInterceptorConfig, getToken, isAuthEnabled, ServerUnavailable} from '@kogito-apps/common';
```
Comment (Sgitario) [inline ui-packages/packages/task-console/src/index.tsx]: Missing the spaces after { and before }

File: ui-packages/packages/task-console/src/index.tsx [hunk 5]
```diff
@@ -3,17 +3,16 @@ import ReactDOM from 'react-dom';
...
 import PageLayout from './components/Templates/PageLayout/PageLayout';
 import TaskConsoleContextProvider from './context/TaskConsoleContext/TaskConsoleContextProvider';
 import taskConsoleLogo from './static/taskConsoleLogo.svg';
+import {setContext} from "apollo-link-context";
```
Comment (Sgitario) [inline ui-packages/packages/task-console/src/index.tsx]: Missing the spaces after { and before }

File: ui-packages/packages/task-console/webpack.common.js [hunk 6]
```diff
@@ -17,7 +17,7 @@ module.exports = {
       KOGITO_AUTH_ENABLED: false,
       KOGITO_KEYCLOAK_REALM: 'kogito',
       KOGITO_KEYCLOAK_URL: 'http://localhost:8280',
-      KOGITO_KEYCLOAK_CLIENT_ID: 'kogito-task-console',
+      KOGITO_KEYCLOAK_CLIENT_ID: 'kogito-console',
```
Comment (Sgitario) [inline ui-packages/packages/task-console/webpack.common.js]: Just a question, as part of this PR, I read kogito-management-console, kogito-task-console, kogito-quarkus-console and kogito-console, what is the difference among them? They are different names for the same compoment?

**Review by Sgitario — COMMENTED** [review]
@nmirasch looks good to me. Just minor thing about naming. 
By the way, what about data-index integration? If the data-index component is configured with security, can we configure management console accordingly? If so, all good, otherwise perhaps we should do this as a separated ticket.

File: ui-packages/packages/task-console/webpack.common.js (hunk 6 above)
Comment (nmirasch) [reply ui-packages/packages/task-console/webpack.common.js]: In this stage I keep two different clients for the consoles kogito-quarkus-console and kogito-console. The difference there is the port that they are able to manage:in  case of kogito-quarkus-console use http://localhost:8380 and the kogito-console use http://localhost:9000. As always, we are open to find a better naming that reflect this :-)

**Comment by nmirasch** [comment]
@Sgitario If the data-index component is configured with security, the service just answers requests with authorization headers with valid keycloak bearer token. Then if you want to connect the console with that data-index you have to enable security to be able to pass that filled headers. When the console have the security enabled it's passing that Authorization header in all its external calls, (to data-index and  to runtimes services)

**Review by Sgitario — APPROVED** [review]

**Comment by nmirasch** [comment]
jenkins please retest this

File: config/kogito-realm.json (hunk 3 above)
Comment (jstastny-cz) [reply config/kogito-realm.json]: it's distinction between kogito-console client for react app and this one which is used when the react app is bundled into quarkus service. Would `kogito-console-quarkus` help?

File: ui-packages/packages/management-console/src/index.html [hunk 7]
```diff
@@ -9,6 +9,10 @@
   </head>
   <script>
     window.DATA_INDEX_ENDPOINT = __DATA_INDEX_ENDPOINT__;
+    window.KOGITO_AUTH_ENABLED = __KOGITO_AUTH_ENABLED__;
+    window.KOGITO_AUTH_KEYCLOAK_REALM =__KOGITO_AUTH_KEYCLOAK_REALM__;
+    window.KOGITO_AUTH_KEYCLOAK_URL=__KOGITO_AUTH_KEYCLOAK_URL__;
+    window.KOGITO_AUTH_KEYCLOAK_CLIENT_ID=__KOGITO_AUTH_KEYCLOAK_CLIENT_ID__;
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/index.html]: whitespace issues in those three lines around `=`

File: ui-packages/packages/task-console/webpack.common.js (hunk 6 above)
Comment (jstastny-cz) [reply ui-packages/packages/task-console/webpack.common.js]: I've already mentioned above, maybe `kogito-console-quarkus` would be a better choice for the latter.

**Comment by nmirasch** [comment]
@jstastny-cz @Sgitario @cristianonicolai  keycloak clients have been renamed to kogito-console-quarkus and kogito-console-react. Let me know if you think there's a better naming for this. Thanks!

**Comment by Sgitario** [comment]
> @jstastny-cz @Sgitario @cristianonicolai keycloak clients have been renamed to kogito-console-quarkus and kogito-console-react. Let me know if you think there's a better naming for this. Thanks!

It sounds pretty much clearer to me. Thanks @nmirasch !

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**